pkg/provider/config.go

```diff
@@ -0,0 +1,46 @@
+package provider
+
+type configuration struct {
+	oidc *configurationOIDC
+	saml *configurationSAML
+}
+
+type configurationOIDC struct {
+	issuer       string
+	keyPath      string
+	clientID     string
+	clientSecret string
+	port         string
+	callbackURL  string
+	scopes       []string
+}
+
+func (c *configurationOIDC) validRS() bool {
+	return c.issuer != "" && c.keyPath != ""
+}
+
+func (c *configurationOIDC) validRP() bool {
+	return c.issuer != "" && c.clientID != "" && c.port != "" && c.callbackURL != ""
+}
```

Comment:
why is there only a single config (authentication and authorization)? e.g. what if i want jwt-profile for both, authentication and authorization?

Reply:
and IMO it's to OIDC focused. I'd go with a `domain` or alike on the provider configuration itself, which represents the zitadel domain to use. IMO easy understandable for devs and regardless of what protocol is used later on it can be used. In case of OIDC this will correspond to the issuer, but the same value can then also be used for the client to call the API and so on

Reply:
If you want a combination then you only have to instantiate multiple providers, as maybe for these 2 functions, only the minimum required values are checked.
Changed the issuer to host, port, and insecure attributes, to concatenate redirectURI and issuer from.